src/rgw/rgw_sync.cc

```diff
@@ -1743,7 +1743,10 @@ class RGWMetaSyncCR : public RGWCoroutine {
           }
         }
         // wait for each shard to complete
-        collect(&ret, NULL);
+        while (ret == 0 && num_spawned() > 0) {
+          yield wait_for_child();
+          collect(&ret, nullptr);
```

Comment:
@cbodley what's the point of collect() here if we don't do anything with `ret`? could just drain_all(), or do something with the error (e.g., log it somewhere).